File: tags/milestone-2-060331/UMLGraph/src/org/daisy/urakawa/exceptions/OperationNotValidException.java
package org.daisy.urakawa.exceptions;

/**
 * This exception should be thrown/raised when trying to
 * call an operation (aka class method) on an object that does not
 * allow a specific modification of the state in the current context.
 * This is an UncheckedException so that not every single low-level
 * operation that potentially modifies the state of the data model
 * has to declare the Exception in its method signature.
 * This allows flexibility in the way the business logic is described
 * in the UML diagram, in terms of the excecution flow (no need for a full state chart).
 * Wherever a "canDoXXX()" method can be found, the corresponding operation "doXXX()"
 * should use this exception/error to let the user-agent of the API/Toolkit
 * know about the non-permitted operation for which there was an attempt to execute.
 */
public class OperationNotValidException extends UncheckedException {
}

File: branches/sdk-core-structedit/exception/OperationNotValidException.cs
using System;

namespace urakawa.exception
{
    /// <summary>
    /// (from the design docs)
    /// This exception should be thrown/raised when trying to
    /// call an operation (aka class method) on an object that does not
    /// allow a specific modification of the state in the current context.
    /// ...
    /// Wherever a "canDoXXX()" method can be found, the corresponding operation "doXXX()"
    /// should use this exception/error to let the user-agent of the API/Toolkit
    /// know about the non-permitted operation for which there was an attempt to execute.
    /// </summary>
    public class OperationNotValidException : UncheckedException
    {
        /// <summary>
        /// Constructor setting the message of the exception
        /// </summary>
        /// <param name="msg">The message</param>
        public OperationNotValidException(string msg) : base(msg)
        {
        }

        /// <summary>
        /// Constructor setting the message and inner <see cref="Exception"/> of the exception
        /// </summary>
        /// <param name="msg">The message</param>
        /// <param name="inner">The inner exception</param>
        public OperationNotValidException(string msg, Exception inner) : base(msg, inner)
        {
        }
    }
}